### DISCUSSION TITLE
Patreon OAuth Provider

### DISCUSSION THREAD

**Discussion by ryceg** [supabase/supabase#7029] [Feature Requests]
Many online content creators use Patreon to pay-gate their premium content. Unfortunately, the Patreon library is poorly documented, and the Patreon team have not updated it in a while. Adding Patreon as a provider would be a great way to bring small content creators to the Supabase platform.

**Comment by namrog84** [comment]
I am working on a new small website and I had planned to specifically use Patreon to pay-gate some premium content. I would love to see this as an integrated auth system.

Either as login using Patreon
or validate an account someother way with an existing patreon account.

**Reply by ryceg** [reply]
Yep- would be awesome to have Patreon working out-of-the-box from Supabase, as the API is pretty terrible.

**Reply by ummahusla** [reply]
I spent a bit of the time reviewing patreon as a way for a pay-gate for a side project, and it would be a dream to have it working out of the box with supabase for sure.

**Comment by fvdsloot** [comment]
Upvoting this for content paygating. Would be amazing!

**Comment by ummahusla** [comment]
Still waiting for this to happen

**Comment by negativems** [comment]
Waiting for this, please make it possible soon

**Comment by martinsix** [comment]
Looks like the internal provider abstraction is pretty simple: https://github.com/supabase/auth/blob/master/internal/api/provider/discord.go

How difficult would it be to create a patreon provider here?

**Comment by Viznity** [comment]
Upvote
